examples/editor/src/App.tsx

```diff
@@ -1,26 +1,162 @@
 // import logo from './logo.svg'
 import "@blocknote/core/style.css";
-import { BlockNoteView, useBlockNote } from "@blocknote/react";
+import {
+  BlockNoteView,
+  getDefaultReactSlashMenuItems,
+  useBlockNote,
+  Theme,
+  lightDefaultTheme,
+  darkDefaultTheme,
+} from "@blocknote/react";
 import styles from "./App.module.css";
+import { BlockSchema, defaultBlockSchema } from "@blocknote/core";
+import { Image, insertImage } from "../../../tests/utils/customblocks/Image";
+import {
+  insertReactImage,
+  ReactImage,
+} from "../../../tests/utils/customblocks/ReactImage";
+import { useEffect, useState } from "react";
 
 type WindowWithProseMirror = Window & typeof globalThis & { ProseMirror: any };
 
-function App() {
-  const editor = useBlockNote({
-    onEditorContentChange: (editor) => {
-      console.log(editor.topLevelBlocks);
+const lightRedTheme = {
+  colors: {
+    editor: {
+      text: "#222222",
+      background: "#ffffff",
+    },
+    menu: {
+      text: "#ffffff",
+      background: "#9b0000",
+    },
+    tooltip: {
+      text: "#ffffff",
+      background: "#b00000",
+    },
+    hovered: {
+      text: "#ffffff",
+      background: "#b00000",
+    },
+    selected: {
+      text: "#ffffff",
+      background: "#c50000",
+    },
+    disabled: {
+      text: "#9b0000",
+      background: "#7d0000",
+    },
+    shadow: "#640000",
+    border: "#870000",
+    sideMenu: "#bababa",
+    // TODO: Update
+    highlightColors: lightDefaultTheme.colors.highlightColors,
+  },
+  borderRadius: 4,
+  fontFamily: "Helvetica Neue, sans-serif",
+} satisfies Partial<Theme>;
+
+const darkRedTheme = {
+  ...lightRedTheme,
+  colors: {
+    ...lightRedTheme.colors,
+    editor: {
+      text: "#ffffff",
+      background: "#9b0000",
     },
-    editorDOMAttributes: {
-      class: styles.editor,
-      "data-test": "editor",
+    sideMenu: "#ffffff",
+    // TODO: Update
+    highlightColors: darkDefaultTheme.colors.highlightColors,
+  },
+} satisfies Partial<Theme>;
+
+const schema = {
+  ...defaultBlockSchema,
+  image: Image,
+  reactImage: ReactImage,
+} satisfies BlockSchema;
+
+function App() {
+  const [theme, setTheme] = useState<
```

Comment:
This should note be here right? if the consumer passes dark + light to the theme prop, blocknote should do the selection?

Reply:
It does, but it needs to also trigger a re-render if the browser theme changes, though this should probably be in `BlockNoteView` either way.